### ISSUE TITLE
Incorrect output filename should have dashes not underscores for tar.gz

### ISSUE THREAD

**Issue by timcu** [pypa/packaging-problems#545]
### OS version

Ubuntu 21.04

### Python version

Python 3.9.5

### Pip version

pip 21.2.4

### Guide link

https://packaging.python.org/tutorials/packaging-projects

### Problem description

Incorrect output filename for tar.gz in your documentation
```
dist/
  example_package_YOUR_USERNAME_HERE-0.0.1-py3-none-any.whl
  example_package_YOUR_USERNAME_HERE-0.0.1.tar.gz
```
Should be
```
dist/
  example_package_YOUR_USERNAME_HERE-0.0.1-py3-none-any.whl
  example-package-YOUR-USERNAME-HERE-0.0.1.tar.gz
```

**Comment by timcu** [comment]
While fixing typos, there is a line which I believe gives the wrong meaning.

The sentence
"\_\_init\_\_.py is required to import the directory as a package, and should be empty."

would be better written as
"\_\_init\_\_.py is required to import the directory as a package, and is empty in this example."

I don't think, if you want to build a package, that "\_\_init\_\_.py" should always be empty.

**Comment by pfmoore** [comment]
> Should be "Uploading `example-package-YOUR-USERNAME-HERE-0.0.1.tar.gz`"

Incorrect. Package names should be normalised to use underscores rather than dashes, as otherwise it's not possible to identify where the name ends and the version begins. See [the specification](https://packaging.python.org/specifications/source-distribution-format/#source-distribution-file-name) for more details.

**Comment by layday** [comment]
This normalisation is not applied to tarballs by setuptools.

**Comment by pfmoore** [comment]
> This normalisation is not applied to tarballs by setuptools.

That's a bug (or less judgmentally, "legacy behaviour" 🙂) in setuptools. I guess if you're describing what actually happens rather than what should happen, then something needs to change, but I'd change the `name` metadata to be `example_package_YOUR_USERNAME_HERE` as that avoids the need to discuss how setuptools differs from the latest standards. It would probably also be worthwhile to change the description of the `name` metadata to suggest only `_` and not `-` (the tutorial already omits the possibility of using `.` in package names, so removing `-` as well shouldn't be a major problem).

**Comment by layday** [comment]
Well, the tutorial is showing what the output should be when you run the command.  In that context, underscores are incorrect.  I don't see a need to discuss why setuptools deviates from the standard in the tutorial - newcomers have no knowledge of the standard and they don't need to know what normalisation rules apply to distribution filenames.  There are also two competing normalisations - the one used for distribution filenames and the one that's used by PyPI.  If people are told to use underscores, they might wonder why PyPI turns them into hyphens.  There's also no guarantee made by the PEP that the un-normalised, underscored name will work, though it obviously does with PyPI.

**Comment by bhrutledge** [comment]
Another potential point of confusion is `example-pkg-YOUR-USERNAME-HERE` vs. `example_package_YOUR_USERNAME_HERE` (changed in https://github.com/pypa/packaging.python.org/commit/57e7d5c56ba62a45efa502b47d2cb200f96e4447). It seems like the former should be un-abbreviated.

It's also a common convention for package names to use hyphens.

> If people are told to use underscores, they might wonder why PyPI turns them into hyphens

@layday I was not familiar with this behavior, but sure enough, https://github.com/pypa/readme_renderer/ is https://pypi.org/project/readme-renderer/.

With all of this in mind, I'm going to transfer this to https://github.com/pypa/packaging.python.org/, with the suggestion that we:

- Finish replacing `pkg` with `package`
- Fix the output to match reality
- Add a note callout about hyphens and underscores
